options.go

```diff
@@ -329,9 +329,12 @@ func Configuration(loader ConfigurationLoader, paths ...string) Option {
 	return OptionFunc(func(k *Kong) error {
 		k.loader = loader
 		for _, path := range paths {
-			if _, err := os.Stat(ExpandPath(path)); os.IsNotExist(err) {
+			f, err := os.Open(ExpandPath(path))
```

Comment:
Thanks for the PR, but the correct fix here is to add `} else { return err }` on the Stat - we don't want to ignore other errors.

Reply:
I'd be happy to accept that change, thanks for tracking this bug down!

Reply:
You mean like this?

```go
			_, err := os.Stat(ExpandPath(path))
			if err != nil {
				if os.IsNotExist(err) {
					continue
				}

				return err
			}
````

This doesn't help, because later when using the resolver there is a panic when an error occurs.

If os.Open fails I think we have to ignore the error, otherwise this leads to a panic later.

Reply:
Why is it panicking later though?

Reply:
Ah I understand now - you should be able to check the result of `Stat()` to determine if the file is readable.

Reply:
I now understand too. The panic occurs here:

```go
	parser, err := New(cli, options...)
	if err != nil {
		panic(err)
	}
```

And if I check in `New` the error appears here:
```go
	for _, option := range options {
		if err := option.Apply(k); err != nil {
			return nil, err
		}
	}
```

On applying the following option:
```go
kong.Configuration(YAML, c.ConfigPaths...),
```

And here the error occures on:

```go
func Configuration(loader ConfigurationLoader, paths ...string) Option {
	return OptionFunc(func(k *Kong) error {
		k.loader = loader
		for _, path := range paths {
			if _, err := os.Stat(ExpandPath(path)); os.IsNotExist(err) {
				continue
			}
			resolver, err := k.LoadConfig(path) <--------------this returns the error
			if err != nil {
				return errors.Wrap(err, path)
			}
			if resolver != nil {
				k.resolvers = append(k.resolvers, resolver)
			}
		}
		return nil
	})
}
```

The code panics before my YAML resolvers is invoked.

It is pretty difficult to check with `Stat` if file is readable for current user. You have to check user,group and other flag.

How about the following:

```go
			f, err := os.Open(ExpandPath(path))
			if err != nil {
				if os.IsNotExist(err) || os.IsPermission(err) {
					continue
				}

				return err
			}

			f.Close()
```

This is pretty explicit what we want to ignore and  only ignores not existing and permission denied errors. What do you think?

Reply:
Yep, great idea - sounds like a good compromise, thank you.

Reply:
I updated the PR. Thx.